WordPress/src/main/java/org/wordpress/android/ui/uploads/UploadService.java

```diff
@@ -51,11 +51,27 @@
     // the post they're attached to, allowing us to update the post with the media URLs in a single batch at the end
     private static final ConcurrentHashMap<Integer, List<MediaModel>> sCompletedMediaByPost = new ConcurrentHashMap<>();
 
+    // This will hold a map of postID and the corresponding last error if any error occured while uploading
+    private static final ConcurrentHashMap<Integer, FailReason> sFailedUploadPosts = new ConcurrentHashMap<>();
+
     @Inject Dispatcher mDispatcher;
     @Inject MediaStore mMediaStore;
     @Inject PostStore mPostStore;
     @Inject SiteStore mSiteStore;
 
+    public class FailReason {
```

Comment:
What about calling this `UploadError` instead? It's going to contain at least one `XError` object (similarly the new methods could be `getUploadErrorForPost()`, `removeUploadErrorForPost()`, etc.